modules/apps/object/object-dynamic-data-mapping/src/main/java/com/liferay/object/dynamic/data/mapping/internal/form/field/type/ObjectRelationshipDDMFormFieldType.java

```diff
@@ -47,7 +47,7 @@
 
 	@Override
 	public String getModuleName() {
-		return "commerce-frontend-js/components/autocomplete/Autocomplete";
+		return "object-dynamic-data-mapping/Autocomplete";
```

Comment:
I don't see how this could really work. This relies on `object-dynamic-data-mapping/Autocomplete` to be aliased and exposed as a global module. This needs to happen via `.npmbridgerc`, which [commerce-frontend-js does](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-frontend-js/.npmbridgerc) and is not really recommended.

As long as you're using a module from the same package, you can use `npmResolver` to get the right path. See:
- [LayoutDDMFormFieldType](https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/layout/dynamic/data/mapping/form/field/type/internal/LayoutDDMFormFieldType.java#L52-L57)
- [JournalArticleDDMFormFieldType](https://github.com/liferay/liferay-portal/blob/master/modules/apps/journal/journal-article-dynamic-data-mapping-form-field-type/src/main/java/com/liferay/journal/article/dynamic/data/mapping/form/field/type/internal/journal/article/JournalArticleDDMFormFieldType.java#L53-L58)

Reply:
@jbalsas is this even the right way to use a js module at all? Brian wanted this autocomplete component moved out of commerce since it was now referenced in `objects`, but I wasn't even sure this was the right way for objects to consume a js component.

Reply:
> @jbalsas is this even the right way to use a js module at all?

Yeah, I think this is legit.

This eventually gets through some layers in ddm until it gets to a `Liferay.Loader.require` call at [`data-engine's Field component`](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/components/Field/Field.es.js#L46-L50)

In our revisited case, the module is accessing an internal component, which doesn't break our principle of just accessing other modules by their main exports. 

In the previous code, objects was accessing commerce's module through a path which should be internals. This only worked because of the bridge mechanism that exposes the whole `commerce-frontend-js` tree. So yeah, accessing external modules by path (digging in their internals) should be persecuted 👊 

I'm not sure if I made a good job at answering, so please, let me know if this is still unclear 😂